V1/Services/OrganisationService.cs

```diff
@@ -122,15 +128,24 @@ public async Task<Response> UpdateCountAsync(string id, UpdateCountRequest paylo
 
         public async Task<Response> MobileEntry(string organisationId, MobileEntryRequest payload)
         {
-            var wallet = await _walletService.CreateWalletAndSms(payload.WalletRequest);
+            var walletRequest = new CreateWalletRequest
+            {
+                MobileNumber = payload.MobileNumber
+            };
+            var wallet = await _walletService.CreateWallet(walletRequest);
+
+            var session = await _sessionService.CreateSession(payload.MobileNumber);
+
+            var organisation = await _organisationRepository.GetAsync(Guid.Parse(organisationId));
+
+            await _smsService.SendMessage(payload.MobileNumber, DefinitionConstants.SmsType.Welcome, organisation.Name, session.ExpireAt.Date);
 
             var updateCounterRequest = new UpdateCountRequest
             {
                 Latitude = payload.Latitude,
                 Longitude = payload.Longitude,
                 WalletId = wallet.Id.ToString()
             };
-
             var counterResponse = await UpdateCountAsync(organisationId, updateCounterRequest, ScanType.CheckIn);
```

Comment:
Rather use the Id linked to the organisation that you got from the repo. IE organisation.Id instead of the variable passed into the method (might not exist)

Reply:
Null check is now added